File: app/src/main/java/com/code_design_camp/client/friday/HeadDisplayClient/ui/MainActivity.java
package com.code_design_camp.client.friday.HeadDisplayClient.ui;


import android.animation.Animator;
import android.content.Intent;
import android.content.SharedPreferences;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.os.Handler;
import android.os.PowerManager;
import android.preference.PreferenceManager;
import android.provider.Settings;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.Window;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.LinearLayout;
import android.widget.ProgressBar;
import android.widget.TextSwitcher;
import android.widget.TextView;
import android.widget.Toast;
import android.widget.ViewFlipper;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.code_design_camp.client.friday.HeadDisplayClient.FridayApplication;
import com.code_design_camp.client.friday.HeadDisplayClient.R;
import com.code_design_camp.client.friday.HeadDisplayClient.Theme;
import com.code_design_camp.client.friday.HeadDisplayClient.activities.FridayActivity;
import com.code_design_camp.client.friday.HeadDisplayClient.fragments.dialogFragments.AuthDialog;
import com.code_design_camp.client.friday.HeadDisplayClient.fragments.dialogFragments.ChangelogDialogFragment;
import com.code_design_camp.client.friday.HeadDisplayClient.fragments.dialogFragments.UninstallOldAppDialog;
import com.code_design_camp.client.friday.HeadDisplayClient.fragments.interfaces.OnAuthCompletedListener;
import com.code_design_camp.client.friday.HeadDisplayClient.fragments.store.MainStoreFragment;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.google.android.material.dialog.MaterialAlertDialogBuilder;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;
import com.google.ar.core.ArCoreApk;
import com.google.ar.core.exceptions.UnavailableDeviceNotCompatibleException;
import com.google.ar.core.exceptions.UnavailableUserDeclinedInstallationException;

public class MainActivity extends FridayActivity {
    public static final int FULLSCREEN_REQUEST_CODE = 22;
    private static final String LOGTAG = "FridayMainActivity";
    //Store Fragment
    MainStoreFragment storeFragment = new MainStoreFragment();
    private ViewFlipper vswitcher_main;

    private AuthDialog authDialogFragment;
    private SharedPreferences defaut_pref;
    private PackageInfo pkgInf;
    private LinearLayout assetLoaderLayout;
    private TextSwitcher assetLoaderText;
    private ProgressBar loadingBar;
    private boolean loadedSpeechRecognizer = false;
    FridayApplication.OnAssetsLoadedListener mAssetsLoadedListener = new FridayApplication.OnAssetsLoadedListener() {
        @Override
        public void onStartedLoadingAssets() {
            assetLoaderLayout.setVisibility(View.VISIBLE);
            Animation anim = AnimationUtils.loadAnimation(MainActivity.this, R.anim.slide_down);
            assetLoaderLayout.startAnimation(anim);
        }

        @Override
        public void onAssetLoaded() {
            loadedSpeechRecognizer = true;
            loadingBar.setIndeterminateDrawable(getDrawable(R.drawable.ic_cloud_done_black_24dp));
            assetLoaderText.setText(getString(R.string.asset_loader_loading_success));
            Handler h = new Handler();
            h.postDelayed(() -> assetLoaderLayout.animate()
                    .alpha(0)
                    .setDuration(500)
                    .setListener(new Animator.AnimatorListener() {
                        @Override
                        public void onAnimationStart(Animator animator) {

                        }

                        @Override
                        public void onAnimationEnd(Animator animator) {
                            assetLoaderLayout.setVisibility(View.GONE);
                        }

                        @Override
                        public void onAnimationCancel(Animator animator) {

                        }

                        @Override
                        public void onAnimationRepeat(Animator animator) {

                        }
                    })
                    .start(), 2000);
        }

        @Override
        public void onError() {
            loadedSpeechRecognizer = false;
            Toast.makeText(MainActivity.this,R.string.err_unable_to_load_speech_assets,Toast.LENGTH_LONG).show();
        }
    };
    OnAuthCompletedListener mOnAuthCompleted;
    BottomNavigationView.OnNavigationItemSelectedListener navselected = new BottomNavigationView.OnNavigationItemSelectedListener() {
        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            switch (item.getItemId()) {
                case R.id.main_nav_dashboard:
                    vswitcher_main.setDisplayedChild(0);
                    break;
                case R.id.main_nav_store:
                    vswitcher_main.setDisplayedChild(2);
                    break;
                case R.id.main_nav_profile:
                    vswitcher_main.setDisplayedChild(1);
                    break;
            }
            return true;
        }
    };
    FloatingActionButton.OnClickListener startVR = view -> {
        Intent intent = new Intent(MainActivity.this, FullscreenActionActivity.class);
        if (loadedSpeechRecognizer && defaut_pref.getBoolean("showHeatWarn", true)) {
            MaterialAlertDialogBuilder builder = new MaterialAlertDialogBuilder(MainActivity.this);
            builder.setTitle("Device health protection")
                    .setMessage("You are running a beta version of friday. This version contains known performance issues causing the device to heat up to an unsafe temperature. To prevent device damage, the AR-activity will be automatically finished after 3 minutes.")
                    .setIcon(R.drawable.ic_warning_black_24dp)
                    .setPositiveButton(android.R.string.ok, (dialogInterface, int_which) -> {
                        startActivityForResult(intent, FULLSCREEN_REQUEST_CODE);
                    })
                    .setNeutralButton("start without limit", (dialogInterface, wich) -> {
                        intent.putExtra("noLimit", true);
                        startActivityForResult(intent, FULLSCREEN_REQUEST_CODE);
                    })
                    .setNeutralButtonIcon(getResources().getDrawable(R.drawable.ic_twotone_security_24px))
                    .setNegativeButton(android.R.string.cancel, null)
                    .create().show();
            return;
        } else if (loadedSpeechRecognizer) {
            startActivity(intent);
            return;
        }
        Snackbar.make(findViewById(R.id.viewflipperparent), R.string.err_assets_not_loaded, Snackbar.LENGTH_SHORT).show();
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        setTheme(Theme.getCurrentAppTheme(this));
        super.onCreate(savedInstanceState);
        supportRequestWindowFeature(Window.FEATURE_NO_TITLE);
        setContentView(R.layout.activity_main);
        Theme theme = new Theme(this);
        int appThemeIndex = theme.indexOf(Theme.getCurrentAppTheme(this));
        findViewById(R.id.mainTitleView).setBackground(theme.createGradient(appThemeIndex));
        findViewById(R.id.profileTitleViewContainer).setBackground(theme.createGradient(appThemeIndex));
        vswitcher_main = findViewById(R.id.main_view_flipper);
        BottomNavigationView main_nav = findViewById(R.id.main_bottom_nav);
        FloatingActionButton lets_go = findViewById(R.id.start_actionmode);
        assetLoaderLayout = findViewById(R.id.asset_loading_indicator_conatiner);
        assetLoaderText = findViewById(R.id.asset_loading_text);
        loadingBar = findViewById(R.id.asset_loading_bar);
        defaut_pref = PreferenceManager.getDefaultSharedPreferences(this);
        FridayApplication app = ((FridayApplication) getApplication());
        try {
            pkgInf = getPackageManager().getPackageInfo(getPackageName(), 0);
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }
        Log.d(LOGTAG, "SharedPref versionName is " + defaut_pref.getString("version", "1.0.0"));
        vswitcher_main.setDisplayedChild(0);
        main_nav.setOnNavigationItemSelectedListener(navselected);
        lets_go.setOnClickListener(startVR);
        checkForFirstUse();
        if (!defaut_pref.getString("version", "0").equals(pkgInf.versionName)|| defaut_pref.getBoolean("pref_devmode_show_changelog", false)) {
            ChangelogDialogFragment changelogdialog = new ChangelogDialogFragment();
            changelogdialog.show(getSupportFragmentManager(), "ChangeLogDialog");
            SharedPreferences.Editor editor = defaut_pref.edit();
            editor.putString("version", pkgInf.versionName);
            editor.apply();
        }
        try {
            PackageManager isOldAppInstalled = getPackageManager();
            isOldAppInstalled.getPackageInfo("com.code_design_camp.client.rasberrypie.rbpieclient", PackageManager.GET_ACTIVITIES);
            UninstallOldAppDialog uninstallOldDialogFragment = new UninstallOldAppDialog();
            getSupportFragmentManager().beginTransaction()
                    .setCustomAnimations(R.anim.slide_up, R.anim.slide_down)
                    .replace(android.R.id.content, uninstallOldDialogFragment)
                    .commit();
        } catch (PackageManager.NameNotFoundException e) {
            Log.i(LOGTAG,"no old version found");
        }
        if (defaut_pref.getInt("theme", 0) == 0) {
            defaut_pref.edit().putInt("theme", R.style.AppTheme).apply();
        }
        PowerManager pm = (PowerManager) getSystemService(POWER_SERVICE);
        assert pm != null;
        if (pm.isPowerSaveMode()) {
            MaterialAlertDialogBuilder builder = new MaterialAlertDialogBuilder(this);
            builder.setTitle(R.string.energy_saver_warn_title);
            builder.setMessage(R.string.energy_saver_warn_msg);
            builder.setPositiveButton(R.string.deactivate, (dialogInterface, i) -> startActivity(new Intent(Settings.ACTION_BATTERY_SAVER_SETTINGS)));
            builder.setNegativeButton(R.string.later, null);
            builder.create().show();
        }
        app.setOnAssetsLoadedListener(mAssetsLoadedListener);
        assetLoaderLayout.setVisibility(View.VISIBLE);
        assetLoaderText.setFactory(() -> new TextView(MainActivity.this));
        assetLoaderText.setInAnimation(AnimationUtils.loadAnimation(this, android.R.anim.fade_in));
        assetLoaderText.setOutAnimation(AnimationUtils.loadAnimation(this, android.R.anim.fade_out));
        assetLoaderText.setText(getString(R.string.asset_loader_loading));
        //Initialize Store
        getSupportFragmentManager().beginTransaction()
                .add(R.id.store_frag_container,storeFragment)
                .commit();
    }

    private void checkForFirstUse() {
        SharedPreferences settingsfile = PreferenceManager.getDefaultSharedPreferences(MainActivity.this);
        if (settingsfile.getBoolean("isFirstUse", true)) {
            MaterialAlertDialogBuilder notifieffirstuse = new MaterialAlertDialogBuilder(MainActivity.this);
            notifieffirstuse.setTitle("Welcome to friday");
            notifieffirstuse.setMessage("Thank you for downloading friday.\n\nRemember that you are in a pre-release of our app - Some features may not work properly or this app will crash at some points.");
            notifieffirstuse.setPositiveButton(android.R.string.ok, null);
            notifieffirstuse.create().show();
            settingsfile.edit().putBoolean("isFirstUse", false).apply();
            Intent showWizard = new Intent(this,WizardActivity.class);
            startActivity(showWizard);
        }
    }

    @Override
    protected void onSaveInstanceState(@NonNull Bundle outState) {
        //Fragments need to be removed before saving the instance state
        getSupportFragmentManager().beginTransaction()
                .remove(storeFragment)
                .commit();
        super.onSaveInstanceState(outState);
    }

    @Override
    public void onResume() {
        super.onResume();
        if (!storeFragment.isAdded()) {
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.store_frag_container, storeFragment)
                    .commit();
        }
    }

    @Override
    public void onBackPressed() {
        if (authDialogFragment.isAdded()) {
            authDialogFragment.dismissDialog();
        } else {
            Snackbar.make(findViewById(R.id.viewflipperparent), getString(R.string.leave_app), Snackbar.LENGTH_SHORT)
                    .setAction(getString(R.string.action_leave), view -> finishAffinity()).show();
        }
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        if (requestCode == FULLSCREEN_REQUEST_CODE && data != null) {
            String errtype = data.getStringExtra("errtype");
            MaterialAlertDialogBuilder alertDialog = new MaterialAlertDialogBuilder(this);
            alertDialog.setPositiveButton(android.R.string.ok, null);
            alertDialog.setNeutralButton(R.string.app_feedback, (dialogInterface, i) -> startActivity(new Intent(MainActivity.this, FeedbackSenderActivity.class)));
            switch (errtype) {
                case "TYPE_NOT_INSTALLED": {
                    alertDialog.setMessage(R.string.errtype_not_installed);
                    ArCoreApk apk = ArCoreApk.getInstance();
                    try {
                        apk.requestInstall(this, true);
                    } catch (UnavailableDeviceNotCompatibleException e) {
                        data.putExtra("errtype","TYPE_DEVICE_NOT_SUPPORTED");
                        onActivityResult(requestCode,resultCode,data);
                    } catch (UnavailableUserDeclinedInstallationException e) {
                        e.printStackTrace();
                    }
                }
                case "TYPE_OLD_APK": {
                    alertDialog.setMessage(R.string.errtype_arcore_apk_too_old);
                }
                case "TYPE_OLD_SDK_TOOL": {
                    alertDialog.setMessage(R.string.errtype_sdk_too_old);
                }
                case "TYPE_DEVICE_INCOMPATIBLE": {
                    alertDialog.setMessage(R.string.errtype_device_incompatible);
                }
            }
            alertDialog.create().show();
        } else {
            super.onActivityResult(requestCode, resultCode, data);
        }
    }

    public void setmOnAuthCompleted(OnAuthCompletedListener mOnAuthCompleted) {
        this.mOnAuthCompleted = mOnAuthCompleted;
        authDialogFragment.setOnAuthListener(mOnAuthCompleted);
    }

    public void promptSignin() {
        Log.d("FirebaseAuth", "showing auth dialog");
        authDialogFragment = new AuthDialog();
        authDialogFragment.setOnAuthListener(mOnAuthCompleted);
        getSupportFragmentManager().beginTransaction()
                .setCustomAnimations(R.anim.slide_up, R.anim.slide_down)
                .replace(android.R.id.content, authDialogFragment)
                .commit();
    }

    public void dismissUninstallPrompt() {
        getSupportFragmentManager().beginTransaction()
                .setCustomAnimations(R.anim.slide_up, R.anim.slide_down)
                .replace(android.R.id.content, new Fragment())
                .commit();
    }

    public AuthDialog getAuthDialogFragment() {
        return authDialogFragment;
    }

    public void goToStore(View v) {
        Intent i = new Intent(this, StoreDetailActivity.class);
        startActivity(i);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.store_default_toolbar, menu);
        return true;
    }
}

File: app/src/main/java/com/friday/ar/ui/mainactivity/MainActivity.kt
package com.friday.ar.ui.mainactivity


import android.Manifest
import android.content.Intent
import android.content.pm.PackageManager
import android.os.Bundle
import android.provider.Settings
import android.util.Log
import android.view.Menu
import android.view.View
import android.view.Window
import android.widget.ImageButton
import android.widget.Toast
import androidx.lifecycle.Observer
import com.crashlytics.android.answers.Answers
import com.crashlytics.android.answers.CustomEvent
import com.friday.ar.FridayApplication
import com.friday.ar.R
import com.friday.ar.base.ui.FullscreenActionActivity
import com.friday.ar.core.Constant
import com.friday.ar.core.Theme
import com.friday.ar.core.activity.FridayActivity
import com.friday.ar.feedback.ui.FeedbackSenderActivity
import com.friday.ar.fragments.dialogFragments.UninstallOldAppDialog
import com.friday.ar.fragments.dialogFragments.UnsupportedDeviceDialog
import com.friday.ar.fragments.dialogFragments.changelog.ChangelogDialogFragment
import com.friday.ar.store.ui.StoreDetailActivity
import com.friday.ar.store.ui.fragments.MainStoreFragment
import com.friday.ar.store.ui.fragments.ManagerBottomSheetDialogFragment
import com.friday.ar.wizard.ui.WizardActivity
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import com.google.android.material.snackbar.Snackbar
import com.google.ar.core.ArCoreApk
import kotlinx.android.synthetic.main.activity_main.*
import org.koin.android.viewmodel.ext.android.viewModel
import java.util.*

@Suppress("unused")
open class MainActivity : FridayActivity() {
    companion object {
        const val FULLSCREEN_REQUEST_CODE = 22
        private const val REQUEST_PERMISSIONS_CODE = 900
        private const val LOGTAG = "FridayMainActivity"
        private const val SITE_DASHBOARD = 0
        private const val SITE_STORE = 1
        private const val SITE_PROFILE = 2
    }

    private var storeFragment = MainStoreFragment()
    internal lateinit var app: FridayApplication
    private val viewModel by viewModel<MainActivityViewModel>()

    private var navselected: BottomNavigationView.OnNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        when (item.itemId) {
            R.id.main_nav_dashboard -> {
                main_view_flipper.displayedChild = SITE_DASHBOARD
                start_actionmode.extend()
            }
            R.id.main_nav_store -> {
                if (findViewById<View>(R.id.stub_page_store) != null) {
                    stub_page_store.visibility = View.VISIBLE
                    setupSorePage()
                }
                main_view_flipper.displayedChild = SITE_STORE
                start_actionmode.shrink()
            }
            R.id.main_nav_profile -> {
                if (findViewById<View>(R.id.stub_page_profile) != null) {
                    stub_page_profile.visibility = View.VISIBLE
                    setupProfilePage()
                }
                main_view_flipper.displayedChild = SITE_PROFILE
                start_actionmode.shrink()
            }
        }
        true
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        supportRequestWindowFeature(Window.FEATURE_NO_TITLE)
        app = application as FridayApplication

        setContentView(R.layout.activity_main)

        if (resources.getBoolean(R.bool.isTablet)) {
            UnsupportedDeviceDialog().show(supportFragmentManager, "UnsupportedDeviceDialog")
            return
        }

        viewModel.getFirstUseState().observe(this, Observer { isFirstUse ->
            if (isFirstUse) {
                val showWizard = Intent(this, WizardActivity::class.java)
                startActivity(showWizard)
            }
        })

        viewModel.getIsUpdatedVersion().observe(this@MainActivity, Observer { isUpdatedVersion ->
            if (isUpdatedVersion) {
                val changeLogDialog = ChangelogDialogFragment()
                changeLogDialog.show(supportFragmentManager, "ChangeLogDialog")
            }
        })

        viewModel.getOldVersionInstalledState().observe(this@MainActivity, Observer { pair ->
            val isOldVersionInstalled = pair.first
            if (isOldVersionInstalled) {
                val dialog = UninstallOldAppDialog()
                val bundle = Bundle()
                Log.d("oldversion", pair.second!!)
                bundle.putString("packageName", pair.second)
                dialog.arguments = bundle
                dialog.show(supportFragmentManager, "UninstallOldAppDialog")
            }
        })

        main_bottom_nav.setOnNavigationItemSelectedListener(navselected)

        start_actionmode.setOnClickListener {
            when {
                viewModel.checkNFC() == MainActivityViewModel.NFCState.NFC_ENABLED -> {
                    val intent = Intent(this@MainActivity, FullscreenActionActivity::class.java)
                    startActivityForResult(intent, FULLSCREEN_REQUEST_CODE)
                    Answers.getInstance().logCustom(CustomEvent(Constant.AnalyticEvent.CUSTOM_EVENT_ACTIONMODE))
                }
                viewModel.checkNFC() == MainActivityViewModel.NFCState.NFC_DISABLED -> {
                    Toast.makeText(this, getString(R.string.armode_requiements_enable_nfc), Toast.LENGTH_LONG).show()
                    startActivity(Intent(Settings.ACTION_NFC_SETTINGS))
                }
                else -> {
                    MaterialAlertDialogBuilder(this)
                            .setTitle(R.string.armode_req_nfc_not_available)
                            .setMessage(R.string.armode_rew_nfc_not_available_msg)
                            .setIcon(R.drawable.ic_nfc_black_24dp)
                            .setPositiveButton(android.R.string.ok) { _, _ -> }
                            .create().show()
                }
            }
        }
    }


    private fun setupSorePage() {
        val storeExpandManagerButton = findViewById<ImageButton>(R.id.storeMore)
        storeExpandManagerButton.setOnClickListener {
            val managerDialog = ManagerBottomSheetDialogFragment()
            managerDialog.show(supportFragmentManager, "ManagerBottomSheet")
        }
    }

    private fun setupProfilePage() {
        val theme = Theme(this@MainActivity)
        findViewById<View>(R.id.profileTitleViewContainer).background = theme.createAppThemeGadient()
    }


    override fun onSaveInstanceState(outState: Bundle) {
        //Fragments need to be removed before saving the instance state
        supportFragmentManager.beginTransaction()
                .remove(storeFragment)
                .commitAllowingStateLoss()
        super.onSaveInstanceState(outState)
    }

    public override fun onResume() {
        super.onResume()
        viewModel.checkForFirstUse()
    }

    override fun onBackPressed() {
        Snackbar.make(findViewById(R.id.viewflipperparent), getString(R.string.leave_app), Snackbar.LENGTH_SHORT)
                .setAction(getString(R.string.action_leave)) { finishAndRemoveTask() }.show()
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        if (requestCode == FULLSCREEN_REQUEST_CODE && data != null) {
            val errtype = data.extras!!.getString("errtype")
            val alertDialog = MaterialAlertDialogBuilder(this)
            alertDialog.setPositiveButton(android.R.string.ok, null)
            alertDialog.setNeutralButton(R.string.app_feedback) { _, _ -> startActivity(Intent(this@MainActivity, FeedbackSenderActivity::class.java)) }
            when (ArCoreApk.Availability.valueOf(errtype!!)) {
                ArCoreApk.Availability.UNKNOWN_ERROR -> alertDialog.setMessage(R.string.unknown_error)
                ArCoreApk.Availability.UNKNOWN_CHECKING -> return
                ArCoreApk.Availability.UNKNOWN_TIMED_OUT -> return
                ArCoreApk.Availability.UNSUPPORTED_DEVICE_NOT_CAPABLE -> alertDialog.setMessage(R.string.errtype_arcore_device_not_capable)
                ArCoreApk.Availability.SUPPORTED_NOT_INSTALLED -> {
                    alertDialog.setMessage(R.string.errtype_not_installed)
                    alertDialog.setPositiveButton(R.string.pluginInstaller_activity_installNow) { _, _ -> ArCoreApk.getInstance().requestInstall(this@MainActivity, true) }
                    alertDialog.setNegativeButton(android.R.string.cancel, null)
                }
                ArCoreApk.Availability.SUPPORTED_APK_TOO_OLD -> alertDialog.setMessage(R.string.errtype_arcore_apk_too_old)
                ArCoreApk.Availability.SUPPORTED_INSTALLED -> alertDialog.setMessage(R.string.errtype_sdk_too_old)
            }
            alertDialog.create().show()
            return
        }
        super.onActivityResult(requestCode, resultCode, data)
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        if (requestCode == REQUEST_PERMISSIONS_CODE) {
            if (!Arrays.equals(grantResults, intArrayOf(PackageManager.PERMISSION_GRANTED, PackageManager.PERMISSION_GRANTED))) {
                val permissionWarnDialog = MaterialAlertDialogBuilder(this)
                permissionWarnDialog.setTitle(R.string.err_missing_permissions)
                        .setMessage(R.string.err_permission_need_explanation)
                        .setIcon(R.drawable.ic_twotone_security_24px)
                        .setCancelable(false)
                        .setPositiveButton(R.string.retry) { _, _ -> requestPermissions(arrayOf(Manifest.permission.CAMERA, Manifest.permission.RECORD_AUDIO), REQUEST_PERMISSIONS_CODE) }
                        .create().show()
            }

        }
    }

    fun goToStore() {
        val i = Intent(this, StoreDetailActivity::class.java)
        startActivity(i)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.store_default_toolbar, menu)
        return true
    }

    fun goToStore(view: View) {}
}
